doc/apiary.apib

```diff
@@ -0,0 +1,754 @@
+                }
+            }
+
++ Response 400 (application/json)
+
+            {
+                "errors": {
+                    "label": [
+                        "Missing data for required field."
+                    ]
+                },
+                "message": "failed to parse input"
+            }
+
+
+### Get the current list of templates [GET]
+Get the full list of templates with all their associated attributes.
+
++ Request
+      + Headers
+
+            Authorization: Bearer JWT
+
++ Response 200 (application/json)
+
+            {
+                "pagination": {
+                    "has_next": false,
+                    "next_page": null,
+                    "page": 1,
+                    "total": 1
+                },
+                "templates": [
+                    {
+                        "attrs": [
+                            {
+                                "created": "2017-12-20T18:42:21.089367+00:00",
+                                "id": 13,
+                                "label": "temperature",
+                                "template_id": "4",
+                                "type": "dynamic",
+                                "value_type": "float"
+                            },
+                            {
+                                "created": "2017-12-20T18:42:21.090601+00:00",
+                                "id": 14,
+                                "label": "pressure",
+                                "template_id": "4",
+                                "type": "dynamic",
+                                "value_type": "float"
+                            },
+                            {
+                                "created": "2017-12-20T18:42:21.091596+00:00",
+                                "id": 15,
+                                "label": "led",
+                                "template_id": "4",
+                                "type": "dynamic",
+                                "value_type": "bool"
+                            },
+                            {
+                                "created": "2017-12-20T18:42:21.092641+00:00",
+                                "id": 16,
+                                "label": "fan",
+                                "template_id": "4",
+                                "type": "dynamic",
+                                "value_type": "bool"
+                            }
+                        ],
+                        "created": "2017-12-20T18:42:21.087782+00:00",
+                        "id": 4,
+                        "label": "SensorModel"
+                    }
+                ]
+            }
+
+
+## Template info [/template/{id}]
+Access a specific template
++ parameters
+    + id: `83af` (required, string) - The template ID
+
+### Delete template [DELETE]
+Removes a template. If any device is based on the template being removed, then all its attributes will be also removed.
```

Comment:
Shouldn't it be deny removing a template that is being used by devices? What if it is the last template of a device? the device is also removed? Or those it become a templateless device?

Reply:
It becomes an attributeless device :)

Reply:
The latter :)
Currently, the device would have its attribute list changed - the attributes from a template are also removed.